imod/util/spatial.py

```diff
@@ -401,6 +417,55 @@ def to_ugrid2d(data: Union[xr.DataArray, xr.Dataset]) -> xr.Dataset:
     return mdal_compliant_ugrid2d(ds)
 
 
+def gdal_compliant_grid(
+    data: Union[xr.DataArray, xr.Dataset],
+    crs: Optional[Any] = None,
+) -> Union[xr.DataArray, xr.Dataset]:
+    """
+    Assign attributes to x,y coordinates to make data accepted by GDAL.
```

Comment:
In nearly all our usecases these attrs are missing. If they are not, attributes are overrided, which I think is fine. This would only be a problem when these attrs would point to different dimensions than y, x. The function fails whenever dimensions "y" and "x" are missing, so this leaves only the tiny tiny corner case where dimensions like "lat" "lon" are included, as well as "y" and "x", and the user is working without validation of their model. In that case I think it's at your own risk ;-)

Reply:
The function is public, and in a general utility namespace, so we should not make too much assumptions about who is going to use it or why. If the data is in a different crs, the x and y coordinate values should be changed if we assign a new crs. This function doesn't do that, and that is fine because doing that is difficult. But throwing an error if you assign a different crs than the grid already has seems useful to do. In any case, if this doesn't happen in our usecases because the grids usually don't have crs'es when they enter this function, then doing this check for a pre-existing crs won't cause problems either.